### ISSUE TITLE
Is JSON:API 1.1 release waiting for implementations?

### ISSUE THREAD

**Issue by bbrala** [json-api/json-api#1578]
From what i understand from the web of issues 1.1 is waiting on enough implementations to get published? Is this correct or is it delayed on some other reason?

**Some implementations i found;**

- https://github.com/woohoolabs/yin - 99% coverage 1.1 they say (server)
- https://github.com/woohoolabs/yang - 99% coverage they say (client)
- https://www.crnk.io/releases/v3.2/ - Adding 1.1. features
- https://github.com/neomerx/json-api - 1.1 specification (used by https://github.com/laravel-json-api, but they dont mention 1.1 unfortunately.

This is what I can find in a quick google session

**Comment by lindyhopchris** [comment]
I was planning to support 1.1 once it was tagged (I'm the author of laravel-json-api).

**Comment by lode** [comment]
I've released a new version for https://github.com/lode/jsonapi adjusting to RC2 & RC3 of v1.1.

**Comment by tobyzerner** [comment]
I'm currently working to implement v1.1 in https://github.com/tobyzerner/json-api-server

**Comment by dgeb** [comment]
v1.1 is primarily waiting for implementations to be finalized, but I don't want this to become a chicken and egg problem. Certainly many implementors will just wait for 1.1 to be finalized before committing to it. However, we don't need more than a couple implementations on each side to finalize this. Thanks for doing some initial research, and thanks to implementors who have already "taken the leap".

A couple more things:
- We will be starting a community chat, [as discussed on twitter](https://twitter.com/jsonapi/status/1424772933705216007). This should help improve communication among implementors and hopefully bring forth some more 1.1 implementations.
- I'd like to convert the site from jekyll to a multi-version [docusaurus](https://docusaurus.io) site. I've hoped to launch this update along with 1.1, but it's not strictly necessary that they be tied together.

**Comment by bbrala** [comment]
Well there seem to be at least a few, isn't that enough yet? ;)

**Comment by dgeb** [comment]
> Well there seem to be at least a few, isn't that enough yet? ;)

Sure, we don't need many. And I wasn't aware of some of these (w/r/t 1.1 support) but I'll check them out soon.

**Comment by bbrala** [comment]
I'm kinda eager to start on 1.1 on drupal, but we can't do much until release as you know :)

**Comment by bradjones1** [comment]
Reference to Drupal being soft-blocked on this: https://www.drupal.org/project/drupal/issues/2955020#comment-13114808

**Comment by bradjones1** [comment]
Related, there is actually a profile already in the docs, but it wasn't linked... this could help demonstrate a model for profile specs published by implementors: https://github.com/json-api/json-api/pull/1582

**Comment by bradjones1** [comment]
Dropping back in here to say that Drupal's ecosystem is basically adopting 1.1 _de facto_; `jsonapi_schema` adds `describedby` links (see https://github.com/json-api/json-api/pull/1447) and `consumer_image_styles` module [includes `rel` in a links object](https://git.drupalcode.org/project/consumer_image_styles/-/merge_requests/1#note_64965) to describe image derivatives. Whether the spec is "official" or not, implementations are popping up to take advantage of its features.

I know everyone is busy - including maintainers. Just providing some extra context as I run into it.

**Comment by bbrala** [comment]
Yeah the Drupal ecosystem is moving forward, core itself needs to be a little more conservative unfortunately. Thanks for the context though @bradjones1

**Comment by auvipy** [comment]
i think this issue https://github.com/json-api/json-api/pull/1600 & another one for v1.1 must also need to be addressed for v1.1 stable

**Comment by bradjones1** [comment]
> Yeah the Drupal ecosystem is moving forward, core itself needs to be a little more conservative unfortunately.

Yes of course; this isn't to co-opt this thread to be about Drupal, but rather address the earlier point about:

> v1.1 is primarily waiting for implementations to be finalized, but I don't want this to become a chicken and egg problem.

The implementations I referenced are server-side, not client-side, but the overarching point is that the nature of APIs in 2022 (isn't that weird to say) is that they are adopting 1.1 conventions _because they need to_. Which is really a good sign that the spec is moving in the right direction by addressing the needs of implementors.

**Comment by freddrake** [comment]
Are the JSON Schema documents considered normative, or are they only intended to be helpful?

Making them normative raises the bar to getting 1.1 done, and raises the possibility of the specification document and the schema being at odds.

I don't think the schema should be normative, though tightly defined schema would definitely be helpful to many developers.

**Comment by bradjones1** [comment]
I'm not a maintainer but I don't see anywhere on the site that indicates it's normative. That said, I think some sort of reference artifact is super helpful. E.g., there's [this issue](https://www.drupal.org/project/drupal/issues/3258084) on Drupal regarding server-side validation of responses.

**Comment by bradjones1** [comment]
Discussed with @gabesullice at DrupalCon last week.

**Comment by freddrake** [comment]
> Discussed with @gabesullice at DrupalCon last week.

And?  Any pending actions or resolutions?

**Comment by bradjones1** [comment]
> And? Any pending actions or resolutions?

I defer to Gabe on what he came away thinking might be the next step. Honestly, I'm not quite sure what anyone but the maintainers (@dgeb and @gabesullice are the two I have spoken to the most) can do but continue to share our experiences in the wild.

I came back here just now to post a link to [this Drupal issue](https://www.drupal.org/project/drupal/issues/3279320#comment-14507478), which illustrates how urgent this is becoming for our implementation, though. We are increasingly shipping functionality that requires 1.1, but can't explicitly support it, which blocks any innovation that isn't critical enough to justify an exception.

Also, the lack of 1.1's official release is blocking the Drupal json:api maintainers from publishing a profile spec. It's very chicken-and-egg.

I certainly don't want to nag in an issue queue. However it does seem to me that many implementations are de-facto implementing 1.1 or postponing new features because it's not published.

**Comment by dgeb** [comment]
I'm sorry for the lack of finality here and the repercussions it has caused in the community. I will commit to finalizing v1.1 by the end of June. There are still [several issues](https://github.com/json-api/json-api/milestone/2) left to address for this milestone, but I believe we can realistically work through them in this time frame.

**Comment by bradjones1** [comment]
Thank you @dgeb! I am going to look now at the milestone queue and see if I can't help with that. Great news!

**Comment by auvipy** [comment]
> I'm sorry for the lack of finality here and the repercussions it has caused in the community. I will commit to finalizing v1.1 by the end of June. There are still [several issues](https://github.com/json-api/json-api/milestone/2) left to address for this milestone, but I believe we can realistically work through them in this time frame.

will be looking forward to it!

**Comment by dgeb** [comment]
Just checking in on the last day of June to update everyone who's anxiously awaiting v1.1 re: the current status. 

First of all, thanks to everyone who's helped to get us quite close to this milestone. Although we've addressed almost all the issues that were on the v1.1 milestone a couple months ago, as well as a number of new ones that have been added since, the last several weeks have brought new discussion and some new work that is still in progress. In particular, landing #1638 now seems quite important to clear up a lot of confusion re: responses. And during the review of this PR, some other issues related to extension compatibility have been raised that I'd like to address. This work will be a high priority in the next few weeks.

After the v1.1 milestone is complete, a final RC will be versioned, which will become the official v1.1 at a pre-determined date if no major flaws or regressions are discovered.

Thanks for your patience and understanding as we navigate this final stretch.

**Comment by auvipy** [comment]
we now have only one issues remain to complete v1.1 O:)

**Comment by dgeb** [comment]
I've just cleared the final issue (besides this one) for the v1.1 milestone. Let's all do a final pass of the spec in the next few days and then move forward. Thanks everyone!

**Comment by dgeb** [comment]
Closing this in favor of the v1.1rc4 tracking issue #1660. If all goes well, v1.1 final will be announced on September 30. Thanks again to everyone for your patience.

**Comment by jelhan** [comment]
Maybe to get back to the title of this issue: v1.1 is not waiting for any implementation. But feedback from implementors before September 30 is highly appreciated!
